scripts/test-e2e-local.js

```diff
@@ -52,105 +56,146 @@ const argv = yargs
-  silent: true,
-})
-  .stdout.trim()
-  .endsWith('-stable');
-
-if (argv.target === 'RNTester') {
-  // FIXME: make sure that the commands retains colors
-  // (--ansi) doesn't always work
-  // see also https://github.com/shelljs/shelljs/issues/86
-  pushd('packages/rn-tester');
+/**
+ * Start the test for RNTester on iOS.
+ *
+ * Parameters:
+ * - @circleCIArtifacts manager object to manage all the download of CircleCIArtifacts. If null, it will fallback not to use them.
+ * - @onReleaseBranch whether we are on a release branch or not
+ */
+async function testRNTesterIOS(circleCIArtifacts, onReleaseBranch) {
+  console.info(
+    `We're going to test the ${
+      argv.hermes ? 'Hermes' : 'JSC'
+    } version of RNTester iOS with the new Architecture enabled`,
+  );
 
-  if (argv.platform === 'iOS') {
-    console.info(
-      `We're going to test the ${
-        argv.hermes ? 'Hermes' : 'JSC'
-      } version of RNTester iOS with the new Architecture enabled`,
+  // remember that for this to be successful
+  // you should have run bundle install once
+  // in your local setup
+  if (argv.hermes && circleCIArtifacts != null) {
+    const hermesURL = await circleCIArtifacts.artifactURLHermesDebug();
+    const hermesPath = path.join(
+      circleCIArtifacts.baseTmpPath(),
+      'hermes-ios-debug.tar.gz',
     );
-
-    // remember that for this to be successful
-    // you should have run bundle install once
-    // in your local setup - also: if I'm on release branch, I pick the
-    // hermes ref from the hermes ref file (see hermes-engine.podspec)
+    // download hermes source code from manifold
+    circleCIArtifacts.downloadArtifact(hermesURL, hermesPath);
+    console.info(`Downloaded Hermes in ${hermesPath}`);
+    exec(
+      `HERMES_ENGINE_TARBALL_PATH=${hermesPath} RCT_NEW_ARCH_ENABLED=1 bundle exec pod install --ansi`,
+    );
+  } else {
     exec(
       `USE_HERMES=${
         argv.hermes ? 1 : 0
       } CI=${onReleaseBranch} RCT_NEW_ARCH_ENABLED=1 bundle exec pod install --ansi`,
     );
+  }
 
-    // if everything succeeded so far, we can launch Metro and the app
-    // start the Metro server in a separate window
-    launchPackagerInSeparateWindow(pwd());
+  // if everything succeeded so far, we can launch Metro and the app
+  // start the Metro server in a separate window
+  launchPackagerInSeparateWindow(pwd());
 
-    // launch the app on iOS simulator
-    exec('npx react-native run-ios --scheme RNTester --simulator "iPhone 14"');
-  } else {
-    // we do the android path here
+  // launch the app on iOS simulator
+  exec('npx react-native run-ios --scheme RNTester --simulator "iPhone 14"');
+}
+
+/**
+ * Start the test for RNTester on Android.
+ *
+ * Parameters:
+ * - @circleCIArtifacts manager object to manage all the download of CircleCIArtifacts. If null, it will fallback not to use them.
+ */
+async function testRNTesterAndroid(circleCIArtifacts) {
+  maybeLaunchAndroidEmulator();
```

Comment:
headsup: `maybeLaunchAndroidEmulator` is broken and it doesn't work. It didn't launch my Android emulator despite no adb devices being connected:
<img width="942" alt="Screenshot 2023-07-03 at 14 30 26" src="https://github.com/facebook/react-native/assets/16104054/a5add4d4-3b79-4ae9-b164-b9f48da88ccc">

Reply:
I haven't touched this... Is it possible that it was already broken?

Reply:
most likely. looks like it was added here https://github.com/facebook/react-native/commit/d631e0d9e950e74576f672f95b86cbfadfd98bf8

Reply:
I think this was broken before. The problem is that the scripts assumes you have `ANDROID_HOME` set or `emulator` is available in your path which might not always be the case